src/systems/systems_wasm/src/lib.rs

```diff
@@ -299,139 +319,275 @@ impl SimulatorReaderWriter for MsfsHandler {
         self.aspects
             .iter_mut()
             .find_map(|aspect| aspect.read(identifier))
-            .unwrap_or(0.)
+            .unwrap_or_else(|| {
+                self.variables
+                    .as_ref()
+                    .map(|registry| registry.read(identifier).unwrap_or(0.))
+                    .unwrap_or(0.)
+            })
     }
 
     fn write(&mut self, identifier: &VariableIdentifier, value: f64) {
         for aspect in self.aspects.iter_mut() {
             if aspect.write(identifier, value) {
-                break;
+                return;
             }
         }
+
+        if let Some(variable_registry) = &mut self.variables {
+            variable_registry.write(identifier, value);
+        }
+    }
+}
+
+/// Declares a variable of a given type with a given name.
+#[derive(Clone)]
+pub enum Variable {
```

Comment:
This is a broad question: We now have the concepts of `Variable`, `VariableType`, `VariableValue`, `VariableIdentifier`. There is quite a bit of code involving `From` and `Into` between these types. Particularly the first three are essentially enums differentiating between aircraft variables, named variables, and aspect variables. Would it not be more feasible to introduce these types (`AircraftVariable`, `NamedVariable`, `AspectVariable`) as structs directly? Shared behavior could be specified via a `Variable` trait perhaps. I think we could avoid some `match` expressions using polymorphism like this.

Reply:
The `AircraftVariable` and `NamedVariable` types are in the `msfs-rs` crate we use. I could of course implement a trait for these types and use these types directly. However, this leads to problems for when I add automated tests, which I'm hoping to do soonish as well.

The main reason for that is that `msfs-rs` only exposes the `AircraftVariable` and `NamedVariable` types when build targeting `wasm32`, which is not what I'd be running the tests with. My hope is that I can flag the current `From<&Variable> for VariableValue` implementation to only compile when targeting `wasm32`, and to create another `From<&Variable> for VariableValue` implementation which would be used for testing, thus removing my dependency on the `msfs-rs` types. If instead I'd use `AircraftVariable` and `NamedVariable` directly, that would be a lot harder, as those types would be scattered throughout.

`AircraftVariable::from` returns a `Result<AircraftVariable, SomeErrror>` which would need to be handled at every usage as well and would likely obscure things.